apidef/oas/operation_test.go

```diff
@@ -1113,3 +1165,295 @@ func verifyOASOperation(t *testing.T, op *openapi3.Operation, method string, cod
+			expectedParts: []pathPart{
+				{name: "users", value: "users", isRegex: false},
+				{name: "profile", value: "profile", isRegex: false},
+			},
+			expectedRegex: false,
+		},
+		{
+			name: "root regex pattern",
+			path: "/.+",
+			expectedParts: []pathPart{
+				{name: "customRegex1", value: ".+", isRegex: true},
+			},
+			expectedRegex: true,
+		},
+		{
+			name: "root regex pattern with trailing slash",
+			path: "/.+",
+			expectedParts: []pathPart{
+				{name: "customRegex1", value: ".+", isRegex: true},
+			},
+			expectedRegex: true,
+		},
+		{
+			name: "path with trailing slash",
+			path: "/users/my-profile/",
+			expectedParts: []pathPart{
+				{name: "users", value: "users", isRegex: false},
+				{name: "my-profile", value: "my-profile", isRegex: false},
+			},
+			expectedRegex: false,
+		},
+		{
+			name: "path with multiple regex patterns",
+			path: "/[0-9]+/[a-zA-Z]+/.*",
+			expectedParts: []pathPart{
+				{name: "customRegex1", value: "[0-9]+", isRegex: true},
+				{name: "customRegex2", value: "[a-zA-Z]+", isRegex: true},
+				{name: "customRegex3", value: ".*", isRegex: true},
+			},
+			expectedRegex: true,
+		},
+		{
+			name:          "path with broken regex",
+			path:          "/[0-9]+/{id:[a-zA-Z+}/.*",
+			expectedParts: []pathPart{},
+			expectedRegex: false,
+		},
+		{
+			name:          "path with unamed broken regex",
+			path:          "/[0-9]+/{[a-zA-Z+}/.*",
+			expectedParts: []pathPart{},
+			expectedRegex: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			parts, hasRegex := splitPath(tt.path)
+
+			if hasRegex != tt.expectedRegex {
+				t.Errorf("splitPath() hasRegex = %v, want %v", hasRegex, tt.expectedRegex)
+			}
+
+			if len(parts) != len(tt.expectedParts) {
+				t.Errorf("splitPath() returned %d parts, want %d", len(parts), len(tt.expectedParts))
+				return
+			}
+
+			for i, part := range parts {
+				expected := tt.expectedParts[i]
+				if part.name != expected.name {
+					t.Errorf("Part %d name = %v, want %v", i, part.name, expected.name)
+				}
+				if part.value != expected.value {
+					t.Errorf("Part %d value = %v, want %v", i, part.value, expected.value)
+				}
+				if part.isRegex != expected.isRegex {
+					t.Errorf("Part %d isRegex = %v, want %v", i, part.isRegex, expected.isRegex)
+				}
+			}
+		})
+	}
+}
+
+func TestValidatePath(t *testing.T) {
+	tests := []struct {
+		name    string
+		path    string
+		wantErr bool
+	}{
+		{
+			name:    "valid simple path",
+			path:    "/api/users",
+			wantErr: false,
+		},
+		{
+			name:    "valid path with parameter",
+			path:    "/api/users/{id}",
+			wantErr: false,
+		},
+		{
+			name:    "valid path with multiple parameters",
+			path:    "/api/users/{id}/posts/{postId}",
+			wantErr: false,
+		},
+		{
+			name:    "valid path with regex parameter",
+			path:    "/api/users/{id:[0-9]+}",
+			wantErr: false,
+		},
+		{
+			name:    "invalid path - unclosed brace",
+			path:    "/api/users/{id",
```

Comment:
Rather you extend some existing tests for httputil.mux with these; if you can, patch the logic there